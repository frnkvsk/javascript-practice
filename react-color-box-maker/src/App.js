/**
 App - this component should render the BoxList component.
 */
import React from "react";
import BoxList from "./BoxList";

import "./App.css";


/** Simple app that just shows the LightsOut game. */

function App() {  
  // let [defaultProps, setProps] = useState("easy");
  // const setDefaultProps = dp => {
  //   defaultProps = dp;
  //   setProps(defaultProps);
  // }
  
  return (
    <div className="App">
      <BoxList />
      
    </div>
  );
}

export default App;
