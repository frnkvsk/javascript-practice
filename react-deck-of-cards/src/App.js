import React from 'react';
import DeckOfCards from './DeckOfCards';
import { makeStyles } from '@material-ui/core/styles';
import { Grid } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: '#c8e6c9',
    width: '100%',
    height: '100vh'
  },
}));

const App = () => {
  const classes = useStyles();
  
  return (
    <Grid container direction="column" className={classes.root} >
      <Grid item container>
        <Grid item xs={"auto"} sm={1} md={2} />
        <Grid item xs={12} sm={10} md={8} >
          <DeckOfCards />
        </Grid>      
        <Grid item xs={"auto"} sm={1} md={2} />
      </Grid>      
    </Grid>
  );
}

export default App;
