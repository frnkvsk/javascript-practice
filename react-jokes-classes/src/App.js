import React from "react";
import JokeListCLS from "./JokeListCLS";

function App() {
  return (
    <div className="App">
      <JokeListCLS />
    </div>
  );
}

export default App;
