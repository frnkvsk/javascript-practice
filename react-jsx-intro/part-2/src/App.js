import React from 'react';
import Tweet from './Tweet'

function App() {
  return (
    <div>
      <Tweet
        name="Harry Henderson"
        username="harryH"
        date={new Date().toDateString()}
        message="I am hairy"
      />
      <Tweet
        name="Mary Mary"
        username="marymary"
        date={new Date().toDateString()}
        message="very merry"
      />
      <Tweet
        name="Anderson Cooper"
        username="andyC"
        date={new Date().toDateString()}
        message="Stay tuned"
      />
    </div>
  );
}

export default App;
