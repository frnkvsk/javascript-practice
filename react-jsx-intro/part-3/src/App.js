import React from 'react';
import Person from './Person';

function App() {
  return (
    <>
    <Person 
      name="Alice"
      age={17}
      hobbies={["golf", "foozball", "painting"]}
    />    
    <Person 
      name="George"
      age={26}
      hobbies={["walking", "running", "jogging"]}
    />    
    <Person 
      name="Anderson"
      age={44}
      hobbies={["Cheeseburgers", "chess", "poker"]}
    />
    </>    
  );
}

export default App;
