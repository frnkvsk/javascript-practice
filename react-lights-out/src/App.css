.App {
  width: 500px;
  text-align: center;
  background: lightgoldenrodyellow;
  border: 1px solid rgb(1, 153, 1, .1);
}
