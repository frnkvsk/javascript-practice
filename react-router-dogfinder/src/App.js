import React, {useState} from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import Nav from './Nav';
import DogList from './DogList';
import DogDetails from './DogDetails';


function App(props) {
  const [name, setName] = useState("");
  const clickHandler = name => {
    setName(name);
  }
  return (
    <div className="App">
      <BrowserRouter>
        <Nav />
        <Switch>
          <Route exact path="/dogs" >
            <DogList props={JSON.stringify(props)} clickHandler={clickHandler}/> 
          </Route>
          <Route path="/dogs/:name" >
            <DogDetails props={JSON.stringify(props.dogs.filter(e => e.name===name)[0])} /> 
          </Route>
          <Redirect to="/dogs" />
        </Switch>
      </BrowserRouter>   
    </div>
  );
}
App.defaultProps = {
  dogs: [
    {
      name: "Whiskey",
      age: 5,
      src: '../images/whiskey.jpg',
      facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
      ]
    },
    {
      name: "Duke",
      age: 3,
      src: '../images/duke.jpg',
      facts: [
        "Duke believes that ball is life.",
        "Duke likes snow.",
        "Duke enjoys pawing other dogs."
      ]
    },
    {
      name: "Perry",
      age: 4,
      src: '../images/perry.jpg',
      facts: [
        "Perry loves all humans.",
        "Perry demolishes all snacks.",
        "Perry hates the rain."
      ]
    },
    {
      name: "Tubby",
      age: 4,
      src: '../images/tubby.jpg',
      facts: [
        "Tubby is really stupid.",
        "Tubby does not like walks.",
        "Angelina used to hate Tubby, but claims not to anymore."
      ]
    }
  ]
}
export default App;
