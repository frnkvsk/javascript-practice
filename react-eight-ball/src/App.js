import React from 'react';
import EightBall from './EightBall';
import ColorBox from './ColorBox';
import './App.css';
function App() {
  return (
    <div className="App">
      <EightBall />
      <ColorBox />
    </div>
  );
}

export default App;
