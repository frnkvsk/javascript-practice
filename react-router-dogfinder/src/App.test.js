import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from './App';

// smoke and appearance test
test('renders dog name from list', () => {
  const { getByText } = render(<App />);
  const listElement = getByText(/Name: Whiskey/i);
  expect(listElement).toBeInTheDocument();
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("handles Nav click event", () => {
  var { getByText } = render(<App />);
  const whiskey = getByText("Whiskey");
  expect(whiskey).toBeInTheDocument();
  fireEvent.click(whiskey);
  var { getByText } = render(<App />);
  const dogFact = getByText("Whiskey loves eating popcorn.");
  expect(dogFact).toBeInTheDocument();

});
it("handles DogList click event", () => {
  var { getByText } = render(<App />);
  const tubby = getByText("Name: Tubby");
  expect(tubby).toBeInTheDocument();
  fireEvent.click(tubby);
  var { getByText } = render(<App />);
  const dogFact = getByText("Tubby is really stupid.");
  expect(dogFact).toBeInTheDocument();

});

it("handles DogDisplay click events", () => {
  var { getByText } = render(<App />);
  const tubby = getByText("Name: Tubby");
  fireEvent.click(tubby);
  var { getByText } = render(<App />);
  const back = getByText(/back/i);
  fireEvent.click(back);
  var { getAllByText } = render(<App />);
  const whiskey = getAllByText("Name: Whiskey");
  expect(whiskey[0]).toBeInTheDocument();

});