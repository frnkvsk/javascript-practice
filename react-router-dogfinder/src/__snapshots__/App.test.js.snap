// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="makeStyles-root-8"
    >
      <a
        class="makeStyles-link-9"
        href="/"
      >
        Dog Home
      </a>
      <a
        class="makeStyles-link-9"
        href="/dogs/Whiskey"
      >
        Whiskey
      </a>
      <a
        class="makeStyles-link-9"
        href="/dogs/Duke"
      >
        Duke
      </a>
      <a
        class="makeStyles-link-9"
        href="/dogs/Perry"
      >
        Perry
      </a>
      <a
        class="makeStyles-link-9"
        href="/dogs/Tubby"
      >
        Tubby
      </a>
    </div>
    <div
      class="makeStyles-root-10"
    >
      <div
        class="makeStyles-imageGallery-11"
      >
        <div
          class="makeStyles-card-12"
        >
          <h3>
            Name: Whiskey
          </h3>
          <img
            alt="Whiskey"
            class="makeStyles-image-13"
            src="../images/whiskey.jpg"
          />
        </div>
        <div
          class="makeStyles-card-12"
        >
          <h3>
            Name: Duke
          </h3>
          <img
            alt="Duke"
            class="makeStyles-image-13"
            src="../images/duke.jpg"
          />
        </div>
        <div
          class="makeStyles-card-12"
        >
          <h3>
            Name: Perry
          </h3>
          <img
            alt="Perry"
            class="makeStyles-image-13"
            src="../images/perry.jpg"
          />
        </div>
        <div
          class="makeStyles-card-12"
        >
          <h3>
            Name: Tubby
          </h3>
          <img
            alt="Tubby"
            class="makeStyles-image-13"
            src="../images/tubby.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
